flink-streaming-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumer.java

```diff
@@ -567,6 +588,76 @@ public void notifyCheckpointComplete(long checkpointId) throws Exception {
 		}
 		return partitionsToSub;
 	}
+
+	/**
+	 * Thread to periodically commit the current read offset into Zookeeper.
+	 */
+	private static class PeriodicOffsetCommitter extends Thread {
+		private long commitInterval;
+		private volatile boolean running = true;
+		private volatile Thread self;
+		private FlinkKafkaConsumer consumer;
+		private final Object stateUpdateLock = new Object();
```

Comment:
You can probably drop this lock, it is used nowhere. If you want to lock, you need to lock on the proper checkpoint lock.

Locking may not be necessary, as these offsets are only some rough inaccurate notion of progress anyways, if checkpointing is not enabled.